### PR TITLE
User-configurable synchronized writes 

### PR DESCRIPTION
**Pull request by adamchester**
**What issue does this PR address?**
#65

**Does this PR introduce a breaking change?**
Yes, it's a breaking change. We *could* make it non-breaking (and I did originally), but it's feels like it might not be worth it. Adding overloads with a new required parameter `syncRoot` just for this seems a little too much.

**Please check if the PR fulfills these requirements**
- [X] The commit follows our [guidelines](https://github.com/serilog/serilog-sinks-console/blob/dev/CONTRIBUTING.md)
- [X] Unit Tests for the changes have been added (for bug fixes / features)

Wit the sample you can:

`dotnet run -- --sync-root-separate` which will produce garbled output:
image

`dotnet run -- --sync-root-same` which produce properly sync'd output:
image

`dotnet run -- --sync-root-default` which will not configure the `syncRoot` (leaving the default as per #65:
image

### CODE REVIEW

File: sample/ConsoleDemo/Program.cs [hunk 1]
```diff
@@ -1,13 +1,63 @@
...
+            System.Threading.Tasks.Parallel.For(0, 1000, options, (i, loopState) =>
+            {
+                var logger = (i % 2 == 0) ? logger1 : logger2;
+                logger.Information("Event {Iteration} generated by {ThreadId}", i, Thread.CurrentThread.ManagedThreadId);
+            });
+
+            Log.CloseAndFlush();
+        }
+
+        static void AnsiConsoleThemeTest()
```
Comment (nblumhardt) [inline sample/ConsoleDemo/Program.cs]: The demo's good, but we should probably keep the example super-simple and revert this file before merging.

File: src/Serilog.Sinks.Console/Sinks/SystemConsole/ConsoleSink.cs [hunk 2]
```diff
@@ -41,10 +42,20 @@ static ConsoleSink()
...
             LogEventLevel? standardErrorFromLevel)
+            : this(theme, formatter, standardErrorFromLevel, _defaultSyncRoot)
+        {
+        }
+
+        public ConsoleSink(
+            ConsoleTheme theme,
+            ITextFormatter formatter,
+            LogEventLevel? standardErrorFromLevel,
+            object syncRoot)
```
Comment (nblumhardt) [inline src/Serilog.Sinks.Console/Sinks/SystemConsole/ConsoleSink.cs]: This class isn't `public`, so we can have just the one, four-parameter constructor, without breaking anyone.

File: sample/ConsoleDemo/Program.cs (hunk 1 above)
Comment (adamchester) [reply sample/ConsoleDemo/Program.cs]: You're right, i'll revert the sample changes.

Do you think it deserves moving the "synchronized writes" sample into a new `csproj` in the sample folder, or deleting is best? It's not particularly easy to demo the issue, that's all.

File: src/Serilog.Sinks.Console/Sinks/SystemConsole/ConsoleSink.cs (hunk 2 above)
Comment (adamchester) [reply src/Serilog.Sinks.Console/Sinks/SystemConsole/ConsoleSink.cs]: Fixed

File: sample/ConsoleDemo/Program.cs (hunk 1 above)
Comment (nblumhardt) [reply sample/ConsoleDemo/Program.cs]: Ah, sorry, missed this - including it as a second sample with a new name sounds good 👍

Comment (adamchester) [reply sample/ConsoleDemo/Program.cs]: added sample in ad280ad

**Comment by adamchester** [comment]
🎉

**Comment by GitSnafu** [comment]
This is great, exactly what I was looking for. Any idea when this might be merged to master?

**Comment by nblumhardt** [comment]
#82 is tracking this

**Pull request merged**